taichi/backends/opengl/opengl_api.cpp

```diff
@@ -21,7 +21,8 @@ namespace opengl {
 #include "taichi/inc/opengl_extension.inc.h"
 #undef PER_OPENGL_EXTENSION
 
-int opengl_threads_per_block = 1024;
+int opengl_max_block_dim = 1024;
+int opengl_max_grid_dim = 1024;
```

Comment:
It seems that we are initializing these variables later using `glGetIntegerv`. Do we still need to specify `1024` here? If we are relying on a later initialization, `0` may be a better value here. 
```suggestion
// opengl_max_block_dim and opengl_max_grid_dim will be initialized later in initialize_opengl
int opengl_max_block_dim = 0;
int opengl_max_grid_dim = 0;
```

Reply:
Just in case `glGetIntegerv` doesn't work properly.